src/test/java/org/dataone/hashstore/HashStoreRunnable.java

```diff
@@ -0,0 +1,78 @@
+package org.dataone.hashstore;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.dataone.hashstore.exceptions.HashStoreServiceException;
+import org.dataone.hashstore.filehashstore.FileHashStoreUtility;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * A HashStoreRunnable represents the data needed for a single request to HashStore
+ * packaged as a Runnable task that can be executed within a thread pool, typically
+ * provided by the Executor service.
+ */
+public class HashStoreRunnable implements Runnable {
+    private static final Log logHashStoreRunnable = LogFactory.getLog(HashStoreRunnable.class);
+    public static final int storeObject = 1;
+    public static final int deleteObject = 2;
+    private final HashStore hashstore;
+    private final int publicAPIMethod;
+    private final String pid;
+    private InputStream objStream;
+
+    public HashStoreRunnable(HashStore hashstore, int publicAPIMethod, InputStream objStream,
+                             String pid) {
+        FileHashStoreUtility.ensureNotNull(hashstore, "hashstore",
+                                           "HashStoreServiceRequestConstructor");
```

Comment:
I still see (and have flagged) more inaccuracies. You're setting your future self up for a lifelong game of whack-a-mole :-)

Try this - I think it solves all your problems, without needing to pass the method name:

```java
// example excerpt, from checkForNotEmptyAndValidString()
//
if (string.isBlank()) {
    StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
    String msg = "Calling Method: " + stackTraceElements[2].getMethodName()
        + "(): argument cannot be empty, etc, etc...";
    throw new IllegalArgumentException(msg);
}
```

[link to original thread, since GH does its best to confuse us](https://github.com/DataONEorg/hashstore-java/pull/58/files/2012a2a728601a1211648e7b226a42ec753b338b#r1708011807)

Reply:
You are beautiful! Thank you for the example excerpt - you have saved future Dou and present Dou appreciates this direction and help.

Reply:
I have also updated all methods that behaved in a similar manner.